sklearn/model_selection/_search.py

```diff
@@ -587,6 +587,10 @@ def fit(self, X, y=None, groups=None, **fit_params):
                               'the "fit" method.', RuntimeWarning)
             else:
                 fit_params = self.fit_params
+        if self.return_train_score == "warn":
+            warnings.warn("return_train_score will default to False from 0.22."
+                          " Please explicitly set return_train_score=True "
+                          "to maintain current behaviour", FutureWarning)
```

Comment:
I think you should mention in the warning that potentially this is going to have a significant performance impact so that return_train_score=True should only be used if you care about the train scores. Please use a better wording than mine ;-).

"Maintain current behaviour" is not great a wording either you should explain very succintly what the current behaviour is.